### PR TITLE
1) Prevent table resizing when input element is added

### PR DESCRIPTION
**Pull request by hiqtodd**
1) Prevent table resizing when input element is added
2) Allow specific cells to be marked as un-editable
3) Prevent duplicate event firing when "Enter" key is pressed

### CODE REVIEW

File: mindmup-editabletable.js [hunk 1]
```diff
@@ -67,7 +70,6 @@ $.fn.editableTableWidget = function (options) {
 			editor.hide();
 		}).keydown(function (e) {
 			if (e.which === ENTER) {
-				setActiveText();
```
Comment (danmaftei) [inline mindmup-editabletable.js:70]: why was this removed? it seems useful to be able to edit a cell by hitting 'enter'

File: mindmup-editabletable.js [hunk 2]
```diff
@@ -10,18 +10,21 @@ $.fn.editableTableWidget = function (options) {
...
 				active = element.find('td:focus');
 				if (active.length) {
-					
-					// Prevent edit of the columns specified. Set columns in options: $('#table').editableTableWidget({ editor: $('<input>'), preventColumns: [ 2, 3 ] });
-					if ($.inArray(active.index() + 1, activeOptions.preventColumns) != -1) {
+					if (
+						// Prevent edit of the columns specified. Set columns in options: $('#table').editableTableWidget({ editor: $('<input>'), preventColumns: [ 2, 3 ] });
+					    activeOptions.preventColumns.includes(active.index()) ||
+						// Prevent edit of specific cells. Add the 'prevent-edit' data attribute to any cell.
+                        $(active).data('preventEdit')
```
Comment (danmaftei) [inline mindmup-editabletable.js]: isn't `active` already a jQuery object? i don't think you need $(active)

File: mindmup-editabletable.js [hunk 3]
```diff
@@ -10,18 +10,21 @@ $.fn.editableTableWidget = function (options) {
...
+			editor = activeOptions.editor.css('position', 'absolute').hide().insertAfter(element.parent()),
 			active,
 			showEditor = function (select) {
 				active = element.find('td:focus');
 				if (active.length) {
-					
-					// Prevent edit of the columns specified. Set columns in options: $('#table').editableTableWidget({ editor: $('<input>'), preventColumns: [ 2, 3 ] });
-					if ($.inArray(active.index() + 1, activeOptions.preventColumns) != -1) {
+					if (
+						// Prevent edit of the columns specified. Set columns in options: $('#table').editableTableWidget({ editor: $('<input>'), preventColumns: [ 2, 3 ] });
```
Comment (danmaftei) [inline mindmup-editabletable.js:19]: fix whitespace

File: mindmup-editabletable.js (hunk 1 above)
Comment (hiqtodd) [reply mindmup-editabletable.js:70]: This is a bug in the code. When the enter key is pressed this also causes the blur event to fire. The table has an event hook on blur to call `setActiveText` so we want to avoid calling it twice.

File: mindmup-editabletable.js (hunk 3 above)
Comment (hiqtodd) [reply mindmup-editabletable.js:19]: The formatting here is fine in Pycharm. I can update it directly through the UI once this is pushed.

File: mindmup-editabletable.js (hunk 2 above)
Comment (hiqtodd) [reply mindmup-editabletable.js]: you're right.
